src/folderlistmodel.cpp

```diff
@@ -333,6 +339,68 @@ int FolderListModel::standardFolderIndex(FolderStandardType folderType)
     return -1;
 }
 
+bool FolderListModel::lessThan(const QMailFolderId &idA, const QMailFolderId &idB)
+{
+    if (!idA.isValid()) {
+        return true;
+    }
+    if (!idB.isValid()) {
+        return false;
+    }
+
+    QMailFolder aFolder(idA), bFolder(idB);
+    if (aFolder.parentFolderId() == bFolder.parentFolderId()) {
+        // Siblings
+        return aFolder.displayName().compare(bFolder.displayName(), Qt::CaseInsensitive) < 0;
+    } else if (aFolder.parentAccountId() != bFolder.parentAccountId()) {
+        // Different accounts, we still want to compare since local storage
+        // can contain some of the standard folders for the account
+        return QMailAccount(aFolder.parentAccountId()).name() < QMailAccount(bFolder.parentAccountId()).name();
+    } else {
+        QMailFolderId commonId;
+        QList<QMailFolderId> aParents;
+        QMailFolderId parentId = idA;
+        while (parentId.isValid()) {
+            QMailFolder folderA(parentId);
+            if (!(folderA.status() & QMailFolder::NonMail)) {
+                aParents.append(parentId);
+            }
+            parentId = folderA.parentFolderId();
+        }
+        if (aParents.contains(idB)) {
+            // b is ancestor of a
+            return false;
+        }
+        QList<QMailFolderId> bParents;
```

Comment:
since you're only using this one to get the last parent, why not QMailFolderId bLastParent and avoid the list overhead?